File: SDKs/velopayments.com/2.26.127/python/src/sdk/models/shared/payeepayorrefv3.py
from dataclasses import dataclass, field
from datetime import date, datetime
from marshmallow import fields
import dateutil.parser
from typing import List,Optional
from enum import Enum
from dataclasses_json import dataclass_json
from sdk import utils
from . import *


@dataclass_json
@dataclass
class PayeePayorRefV3:
    invitation_status: Optional[InvitationStatus2Enum] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invitationStatus') }})
    invitation_status_timestamp: Optional[datetime] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('invitationStatusTimestamp'), 'encoder': utils.datetimeisoformat(True), 'decoder': dateutil.parser.isoparse, 'mm_field': fields.DateTime(format='iso') }})
    payable_issues: Optional[List[PayableIssue]] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('payableIssues') }})
    payable_status: Optional[bool] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('payableStatus') }})
    payment_channel_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('paymentChannelId') }})
    payor_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('payorId') }})
    remote_id: Optional[str] = field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('remoteId') }})
    

File: SDKs/velopayments.com/2.26.127/typescript/src/sdk/models/shared/payeepayorrefv3.ts
import { SpeakeasyMetadata, SpeakeasyBase } from "../../../internal/utils";
import { InvitationStatus2Enum } from "./invitationstatus2enum";
import { PayableIssue } from "./payableissue";



export class PayeePayorRefV3 extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=invitationStatus" })
  invitationStatus?: InvitationStatus2Enum;

  @SpeakeasyMetadata({ data: "json, name=invitationStatusTimestamp" })
  invitationStatusTimestamp?: Date;

  @SpeakeasyMetadata({ data: "json, name=payableIssues", elemType: PayableIssue })
  payableIssues?: PayableIssue[];

  @SpeakeasyMetadata({ data: "json, name=payableStatus" })
  payableStatus?: boolean;

  @SpeakeasyMetadata({ data: "json, name=paymentChannelId" })
  paymentChannelId?: string;

  @SpeakeasyMetadata({ data: "json, name=payorId" })
  payorId?: string;

  @SpeakeasyMetadata({ data: "json, name=remoteId" })
  remoteId?: string;
}
